code/game/machinery/computer/cargo.dm

```diff
@@ -0,0 +1,528 @@
+/* Contents:
+ - supplycomp (This one approves orders)
+ - ordercomp (This is the public-facing one)
+For the shuttle controller, see supplyshuttle.dm
+For cargo crates, see supplypacks.dm
+For vending packs, see vending_packs.dm*/
+
+// returns an associate list of information needed for cargo consoles.  returns 0 if ID or account is missing
+/proc/get_account_info(mob/user)
+	var/list/acc_info = new
+	var/obj/item/weapon/card/id/usr_id = user.get_id_card()
+	if(ishuman(user))
+		if(usr_id == null)
+			to_chat(user, "<span class='warning'>Please wear an ID with an associated bank account.</span>")
+			return 0
+		acc_info["idname"] = usr_id.registered_name
+		acc_info["idrank"] = usr_id.GetJobName()
+	else if(issilicon(user))
+		acc_info["idname"] = user.real_name
+		acc_info["idrank"] = "Cyborg"
+	var/datum/money_account/account = user.get_worn_id_account()
+	if(!account)
+		to_chat(user, "<span class='warning'>Please wear an ID with an associated bank account.</span>")
+		return 0
+	acc_info["account"] = account
+	return acc_info
+
+/obj/machinery/computer/supplycomp
+	name = "Supply shuttle console"
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "supply"
+	req_access = list(access_cargo)
+	circuit = "/obj/item/weapon/circuitboard/supplycomp"
+	var/reqtime = 0 //Cooldown for requisitions - Quarxink
+	var/hacked = 0
+	var/can_order_contraband = 0
+	var/permissions_screen = FALSE
+	var/last_viewed_group = "Supplies" // not sure how to get around hard coding this
+	var/datum/money_account/current_acct
+
+	light_color = LIGHT_COLOR_BROWN
+
+/obj/machinery/computer/supplycomp/attack_ai(var/mob/user as mob)
```

Comment:
this should already be handled in `/obj/machinery/attack_ai`

Reply:
I am sure that there are a lot of problems with the supplycomp that were already there, it's oldcode and has a lot of implementation problems (not the least of which being the copypaste code in the orderconsole's doorder segment of Topic).

However I do not believe it is in-scope to rewrite the entire file to modern standards just because I'm splitting it into a new file.